app/src/main/java/com/firebase/uidemo/auth/SignedInActivity.java

```diff
@@ -174,20 +177,33 @@ private void populateProfile() {
     }
 
     private void populateIdpToken() {
-        IdpResponse idpResponse = getIntent().getParcelableExtra(EXTRA_IDP_RESPONSE);
+        IdpResponse idpResponse = getIntent().getParcelableExtra(ExtraConstants.EXTRA_IDP_RESPONSE);
+        String token = null;
+        String secret = null;
         if (idpResponse != null) {
-            String token = idpResponse.getIdpToken();
-            String secret = idpResponse.getIdpSecret();
-            if (token == null) {
-                findViewById(R.id.idp_token_layout).setVisibility(View.GONE);
-            } else {
-                ((TextView) findViewById(R.id.idp_token)).setText(token);
-            }
-            if (secret == null) {
-                findViewById(R.id.idp_secret_layout).setVisibility(View.GONE);
-            } else {
-                ((TextView) findViewById(R.id.idp_secret)).setText(secret);
-            }
+            token = idpResponse.getIdpToken();
+            secret = idpResponse.getIdpSecret();
+        }
+        if (token == null) {
+            findViewById(R.id.idp_token_layout).setVisibility(View.GONE);
+        } else {
+            ((TextView) findViewById(R.id.idp_token)).setText(token);
+        }
+        if (secret == null) {
+            findViewById(R.id.idp_secret_layout).setVisibility(View.GONE);
+        } else {
+            ((TextView) findViewById(R.id.idp_secret)).setText(secret);
+        }
+    }
```

Comment:
I did some cleanup here so that if `IdpResponse` is `null`, we don't show the layouts.